src/Corruptions.ts

```diff
@@ -1,12 +1,348 @@
 import i18next from 'i18next'
-import { DOMCacheGetOrSet, DOMCacheHas } from './Cache/DOM'
-import { PCoinUpgradeEffects } from './PseudoCoinUpgrades'
+import { DOMCacheGetOrSet } from './Cache/DOM'
 import { format, player } from './Synergism'
 import { IconSets } from './Themes'
 import { toggleCorruptionLevel } from './Toggles'
 import { Alert, Prompt } from './UpdateHTML'
-import { getElementById } from './Utility'
+import { getElementById, productContents, sumContents, validateNonnegativeInteger } from './Utility'
 import { Globals as G } from './Variables'
+import { PCoinUpgradeEffects } from './PseudoCoinUpgrades'
+
+export const convertInputToCorruption = (array: number[]): Corruptions => {
+  return {
+    viscosity: array[0],
+    drought: array[6],
+    deflation: array[4],
+    extinction: array[5],
+    illiteracy: array[3],
+    recession: array[7],
+    dilation: array[1],
+    hyperchallenge: array[2]
+  }
+}
+
+export type Corruptions = {
+  viscosity: number
+  drought: number
+  deflation: number
+  extinction: number
+  illiteracy: number
+  recession: number
+  dilation: number
+  hyperchallenge: number
+}
+
+export const c15Corruptions: Corruptions = {
+  viscosity: 11,
+  drought: 11,
+  deflation: 11,
+  extinction: 11,
+  illiteracy: 11,
+  recession: 11,
+  dilation: 11,
+  hyperchallenge: 11
+}
+
+export class CorruptionLoadout {
+  #totalScoreMult = 1
+  #corruptionScoreMults = [1, 3, 4, 5, 6, 7, 7.75, 8.5, 9.25, 10, 10.75, 11.5, 12.25, 13, 16, 20, 25, 33, 35]
+  #levels: Corruptions = {
+    viscosity: 0,
+    drought: 0,
+    deflation: 0,
+    extinction: 0,
+    illiteracy: 0,
+    recession: 0,
+    dilation: 0,
+    hyperchallenge: 0
+  }
+  #bonusLevels = 0
+
+  constructor (p: Partial<Corruptions>) {
+    Object.assign(this.#levels, p)
+  }
```

Comment:
I don't there there is ever a reason to use Partial. Whatever is being passed here should be populated with empty values instead.